### PR TITLE
Remove `pgdb` from `envtool`

### LINKED ISSUE #3452: Remove `pgdb` from `envtool`
### What should be done?

`envtool` should use the new PostgreSQL backend (probably, `metadata/pool` package), not `pgdb`.

### Where?

https://github.com/FerretDB/FerretDB/tree/main/cmd/envtool

### Definition of Done

- `envtool` no longer uses `pgdb`;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by ShatilKhan**
# Description
Removed `pgdb` from `envtool.go` file 

Closes #3452 

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [ ] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [X] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [X] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: cmd/envtool/envtool.go [hunk 1]
```diff
@@ -98,30 +97,8 @@ func setupAnyPostgres(ctx context.Context, logger *zap.SugaredLogger, uri string
...
-	}
-
-	if ctx.Err() != nil {
-		return ctx.Err()
-	}
-
-	defer pgPool.Close()
-
-	return nil
-}
```
Comment (AlekSi) [inline cmd/envtool/envtool.go:124]: That code waited for PostgreSQL to start and checked connection and authentication. We still need to do that, but by using new backend code.

**Review by AlekSi — COMMENTED** [review]
Please click the CLA button in that comment: https://github.com/FerretDB/FerretDB/pull/3586#issuecomment-1764112881

File: cmd/envtool/envtool.go (hunk 1 above)
Comment (ShatilKhan) [reply cmd/envtool/envtool.go:124]: Thank you for quick response @AlekSi  & for the explanation 
I see in the issue you stated that we probably need to use `metadata/pool` package
Can give more Information on that?
And lastly I'm supposed to work on only that section of the code you mentioned?

Comment (AlekSi) [reply cmd/envtool/envtool.go:124]: > Can you give more Information on that?

https://github.com/FerretDB/FerretDB/blob/main/internal/backends/postgresql/metadata/pool/pool.go
Create a new pool, call `Pool.Get` until it stops returning an error, close the pool.

> And lastly I'm supposed to work on only that section of the code you mentioned?

Yes, we just need to replace that part of the code

File: cmd/envtool/envtool.go [hunk 2]
```diff
@@ -98,13 +98,11 @@ func setupAnyPostgres(ctx context.Context, logger *zap.SugaredLogger, uri string
...
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/3452
-
-	var pgPool *pgdb.Pool
+	var pgPool *pool.Pool
 
 	var retry int64
 	for ctx.Err() == nil {
-		if pgPool, err = pgdb.NewPool(ctx, uri, logger.Desugar(), sp); err == nil {
+		if pgPool, err = pool.New(uri, logger.Desugar(), sp); err == nil {
```
[the remark this answers is not in the document]
Comment (chilagrow) [reply cmd/envtool/envtool.go:105]: @noisersup, let's check connection.

File: cmd/envtool/envtool.go [hunk 3]
```diff
@@ -98,16 +99,20 @@ func setupAnyPostgres(ctx context.Context, logger *zap.SugaredLogger, uri string
...
 
 	var retry int64
 	for ctx.Err() == nil {
-		if pgPool, err = pgdb.NewPool(ctx, uri, logger.Desugar(), sp); err == nil {
+		var p *pool.Pool
+		if p, err = pool.New(uri, logger.Desugar(), sp); err == nil {
 			break
 		}
 
+		pgPool, err = p.Get("username", "password")
```
Comment (chilagrow) [inline cmd/envtool/envtool.go:111]: Shouldn't we get authentication from `uri`?

File: cmd/envtool/envtool.go [hunk 4]
```diff
@@ -98,16 +99,20 @@ func setupAnyPostgres(ctx context.Context, logger *zap.SugaredLogger, uri string
...
-	// TODO https://github.com/FerretDB/FerretDB/issues/3452
-
-	var pgPool *pgdb.Pool
+	var pgPool *pgxpool.Pool
 
 	var retry int64
 	for ctx.Err() == nil {
-		if pgPool, err = pgdb.NewPool(ctx, uri, logger.Desugar(), sp); err == nil {
+		var p *pool.Pool
+		if p, err = pool.New(uri, logger.Desugar(), sp); err == nil {
```
Comment (chilagrow) [inline cmd/envtool/envtool.go:107]: If `err == nil`, it breaks in L108. I don't think that's what we want, we want L111 to execute.
Could you add simple test please? 🤗

Comment (noisersup) [reply cmd/envtool/envtool.go]: nice catch

**Review by chilagrow — DISMISSED** [review]
Thank you for this :hugs:

File: cmd/envtool/envtool.go [hunk 5]
```diff
@@ -98,13 +104,20 @@ func setupAnyPostgres(ctx context.Context, logger *zap.SugaredLogger, uri string
...
-
-	var pgPool *pgdb.Pool
+	var pgPool *pgxpool.Pool
 
 	var retry int64
 	for ctx.Err() == nil {
-		if pgPool, err = pgdb.NewPool(ctx, uri, logger.Desugar(), sp); err == nil {
+		var p *pool.Pool
+
+		if p, err = pool.New(uri, logger.Desugar(), sp); err != nil {
```
Comment (AlekSi) [inline cmd/envtool/envtool.go:113]: Why do we recreate a pool in a loop? And why do we never close it?

**Review by AlekSi — APPROVED** [review]

**Pull request merged**